lib/bolt/pal.rb

```diff
@@ -143,6 +143,33 @@ def list_tasks
       end
     end
 
+    def parse_param(str, type, compiler)
+      begin
+        parsed = JSON.parse(str, quirks_mode: true)
+        type.instance?(parsed) ? parsed : str
+      rescue JSON::ParserError
+        # This value may not be assignable in which case run_* will error
+        str
+      end
+    end
+
+    def parse_task_params(task_name, params, compiler)
+      sig = compiler.task_signature(task_name)
+      # CODEREVIEW: How should we handle the lack of a task signature?
+      # This will primarily be an issue for non-local orch tasks
```

Comment:
The best option I've come up with is to fallback on querying Orchestrator if: no local task is found, and Orchestrator is configured. That doesn't seem ideal though.

Reply:
I think interacting with the transport API it too rickety.
We should support a remote task store for using orchestrator and use that for validation.